src/pages/SignupHandler/index.js

```diff
@@ -15,236 +15,269 @@ import { routerRedux } from 'dva/router';
 import { connect } from 'dva';
 import styles from './index.less';
 
-@connect(auth => ({
-  auth: auth.auth,
+@connect(({ datastore }) => ({
+  datastoreConfig: datastore.datastoreConfig,
 }))
 class SignupHandler extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      firstName: '',
+      lastName: '',
       username: '',
       password: '',
-      confPassword: '',
-      email: '',
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-      charLength: ['block', 'none'],
-      specialChar: ['block', 'none'],
-      number: ['block', 'none'],
-      upperCaseChar: ['block', 'none'],
-      passwordMatch: '',
-      emailValidation: '',
+      // eslint-disable-next-line react/no-unused-state
+      confirmedPassword: '',
+      submitButtonDisabled: true,
+      submitButtonColor: 'black',
+      passwordValidation: {
+        charCount: false,
+        specialCharCount: false,
+        containsNumber: false,
+        containsUppercaseLetter: false,
+      },
+      passwordValidationMessage: '',
+      usernameValidationMessage: '',
     };
   }
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
-  };
+  componentDidMount() {
```

Comment:
OK, I've used Javascript but I'm hardly an expert. I assume that converting this from a lambda binding into a function must be more than just stylistic, especially considering that the real implementation is now just another lambda binding. Why?